backend/src/target.ts

```diff
@@ -133,8 +130,8 @@ class Target {
     });
   }
 
-  async getFeedId(signer: AddressOrPair): Promise<U64> {
-    const { address } = (signer as KeyringPair);
+  async getFeedId(signer: SignerWithAddress): Promise<U64> {
```

Comment:
since we're not casting anymore I would suggest: ` async getFeedId({ address }: SignerWithAddress): Promise<U64> {`

Reply:
We still need full signed further in the function

Reply:
right, my bad